src/historywork/WriteSnapshotWork.cpp

```diff
@@ -12,9 +12,13 @@
 namespace stellar
 {
 
+// Note, WriteSnapshotWork does not have any special retry clean-up logic:
+// history items are written via XDROutputFileStream, which automatically
+// truncates any existing files.
 WriteSnapshotWork::WriteSnapshotWork(Application& app,
                                      std::shared_ptr<StateSnapshot> snapshot)
-    : BasicWork(app, "write-snapshot", Work::RETRY_A_LOT), mSnapshot(snapshot)
+    : BasicWork(app, "write-snapshot", BasicWork::RETRY_A_FEW)
```

Comment:
I also do not see a reason to not retry a lot here

Reply:
I think `PublishWork` doesn't even need to retry: it's designed to be rescheduled "forever"

Reply:
For `WriteSnapshotWork`: the only time it actually - transiently - fails is due to read/write transaction serialization order:
https://github.com/stellar/stellar-core/blob/89e994b5cffa304eb5449e40cc9fc7a35c205b2d/src/history/StateSnapshot.cpp#L138

I don't feel particularly strongly about retrying a lot here, it just seems like retrying 5 times with exponential backoff should be enough.